docs/sources/panels-visualizations/visualizations/flame-graph/index.md

```diff
@@ -19,61 +19,115 @@ weight: 100
+You can use a flame graph visualization if you need to:
+
+- Identify any performance hotspots to find where code optimizations may be needed.
+- Diagnose the root cause of any performance degradation.
+- Analyze the behavior of complex systems, including distributed systems or microservices architectures.
+
+## Configure a flame graph visualization
+
+Once you’ve created a [dashboard](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/dashboards/build-dashboards/create-dashboard/), the following video shows you how to configure a flame graph visualization:
+
+<!-- video TBA here -->
+
+## Supported data formats
 
-![<Flame graph>](/media/docs/grafana/panels-visualizations/flamegraph/screenshot-flamegraph-10.1-flamegraph.png)
+To render a flame graph, you must format the data frame data using a _nested set model_.
+
+A nested set model ensures each item of a flame graph is encoded by its nesting level as an integer value, its metadata, and by its order in the data frame. This means that the order of items is significant and needs to be correct. The ordering is a depth-first traversal of the items in the flame graph which recreates the graph without needing variable-length values in the data frame like in a children's array.
+
+Required fields:
+
+| Field name | Type   | Description                                                                                                                |
+| ---------- | ------ | -------------------------------------------------------------------------------------------------------------------------- |
+| level      | number | The nesting level of the item. In other words how many items are between this item and the top item of the flame graph.    |
+| value      | number | The absolute or cumulative value of the item. This translates to the width of the item in the graph.                       |
+| label      | string | Label to be shown for the particular item.                                                                                 |
+| self       | number | Self value, which is usually the cumulative value of the item minus the sum of cumulative values of its immediate children. |
+
+### Example
+
+The following table is an example of the type of data you need for a flame graph visualization and how it should be formatted:
+
+#### Time-series table
+
+| level | value    | self   | label                                     |
+| ----- | -------- | ------ | ----------------------------------------- |
+| 0     | 16.5 Bil | 16.5 K | total                                     |
+| 1     | 4.10 Bil | 4.10 k | test/pkg/agent.(\*Target).start.func1     |
+| 2     | 4.10 Bil | 4.10 K | test/pkg/agent.(\*Target).start.func1     |
+| 3     | 3.67 Bil | 3.67 K | test/pkg/distributor.(\*Distributor).Push |
+| 4     | 1.13 Bil | 1.13 K | compress/gzip.(\*Writer).Write            |
+| 5     | 1.06 Bil | 1.06 K | compress/flate.(\*compressor).write       |
```

Comment:
Is the word "flate" correct here? It's causing an error so I just want to make sure.
```suggestion
| 5     | 1.06 Bil | 1.06 K | compress/flat.(\*compressor).write       |
```